### PR TITLE
(wip) minimal change

### PR DESCRIPTION
**Pull request by BadSingleton**
still crashes though

### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Checklist

Check all those that are applicable and complete.

-   [ ] Make sure to include one or more tests for your change
-   [ ] If an enhancement PR, please create docs and at best an example
-   [ ] Add yourself to [`AUTHORS`](../blob/master/AUTHORS.md)
-   [ ] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

File: src/runtime/classmanager.cs [hunk 1]
```diff
@@ -82,30 +82,31 @@ private static int TraverseTypeClear(IntPtr ob, IntPtr arg)
...
-            storage.AddValue("cache", cache);
-            foreach (var cls in cache.Values)
-            {
-                // This incref is for cache to hold the cls,
-                // thus no need for decreasing it at RestoreRuntimeData.
-                Runtime.XIncref(cls.pyHandle);
-                var context = contexts[cls.pyHandle] = new InterDomainContext();
-                cls.Save(context);
-            }
+            // var contexts = storage.AddValue("contexts",
```
Comment (BadSingleton) [inline src/runtime/classmanager.cs]: The only thing saved here is the cache, and we don't want to. The cache is actually cleared in Runtime.Shutdown()

File: src/runtime/managedtype.cs [hunk 2]
```diff
@@ -246,6 +246,13 @@ protected static void ClearObjectDict(IntPtr ob)
             Runtime.XDecref(dict);
         }
 
+        public override string ToString()
```
Comment (BadSingleton) [inline src/runtime/managedtype.cs:249]: Mostly a debug helper

File: src/runtime/typemanager.cs [hunk 3]
```diff
@@ -70,17 +71,28 @@ internal static void SaveRuntimeData(RuntimeDataStorage storage)
...
-            foreach (var entry in cache)
+            var _cache = new Dictionary<MaybeType, IntPtr>();
+            storage.GetValue("cache", out _cache);
+            foreach (var entry in _cache)
             {
-                Type type = entry.Key;
+                Type type = null;
+                try
+                {
+                    type = entry.Key.Value;
```
Comment (benoithudson) [inline src/runtime/typemanager.cs:88]: This argues for a HasValue so you can write `if (entry.Key.HasValue)`

Comment (benoithudson) [reply src/runtime/typemanager.cs:88]: Oh you already have Valid

**Pull request closed without merging**